plugins/module_utils/network/sonic/config/l3_interfaces/l3_interfaces.py

```diff
@@ -256,521 +296,491 @@ def _state_deleted(self, want, have):
         :returns: the commands necessary to remove the current configuration
                   of the provided objects
         """
-        commands = list()
-        if not want:
-            commands = self.remove_default_entries(have)
-            requests = self.get_delete_all_completely_requests(commands)
+        commands, requests = [], []
+        if not have:
+            return commands, requests
+        elif not want:
+            for have_conf in have:
+                commands.append({'name': have_conf['name']})
+                requests.extend(self.get_delete_l3_interface_requests(have_conf, have_conf))
         else:
-            commands = self.remove_default_entries(want)
-            filtered_have = self.remove_default_entries(have)
-            requests = self.get_delete_l3_interfaces_requests(commands, filtered_have)
-        if len(requests) == 0:
-            commands = []
+            for conf in want:
+                intf_name = conf['name']
+                have_conf = next((item for item in have if item['name'] == intf_name), None)
```

Comment:
Here also dict comprehension could be used to reduce the time spent to determine "have_conf". (REF: the comment above on lines 194-197.)